kernel/src/main/java/org/kframework/compile/AddSortInjections.java

```diff
@@ -118,14 +136,38 @@ public K addInjections(K term, Sort expectedSort) {
         }
     }
 
+    private Context addInjections(Context context) {
+        return new Context(addSortInjections(context.body(), Sorts.K()), addSortInjections(context.requires(), Sorts.Bool()), context.att());
+    }
+
+    private Rule addInjections(Rule rule) {
+        return new Rule(addTopSortInjections(rule.body()), addSortInjections(rule.requires(), Sorts.Bool()), addSortInjections(rule.ensures(), Sorts.Bool()), rule.att());
+    }
+
     private K visitChildren(K term, Sort parentSort, Sort actualSort) {
         Att att = term.att().add(Sort.class, actualSort);
         if (term instanceof KApply) {
             KApply kapp = (KApply)term;
             if (kapp.klabel().name().equals("inj")) {
                 return term;
-            } else if (kapp.klabel().name().equals("#Or")) {
-                return KApply(KLabel(kapp.klabel().name(), actualSort), KList(addInjections(kapp.items().get(0), actualSort), addInjections(kapp.items().get(1), actualSort)), att);
+            } else if (kapp.klabel().name().equals("#True")
```

Comment:
I don't like this. This is what the generic poly machinery should be for.

Reply:
I manage to reduce it to the symbols which can have any output sort (True/False/Ceil/Floor/Equals).  I'm not sure how to reduce it further but we can talk about that.